src/Turtle/Prelude.hs

```diff
@@ -1090,6 +1093,32 @@ time io = do
 hostname :: MonadIO io => io Text
 hostname = liftIO (fmap Text.pack getHostName)
 
+-- | Show the full path of an executable file
+which :: MonadIO io => FilePath -> io (Maybe FilePath)
+which cmd = fold go Control.Foldl.head
+  where
+    go :: Shell FilePath
+    go =
+      need "PATH" >>= \case
+          Nothing -> empty
+          Just paths -> do
+            path <- select (Text.split (== ':') paths)
+            let path' = Filesystem.fromText path </> cmd
+            exists <- testfile path'
+            if exists
+                then do
+                    perms <- liftIO $
+                        getmod path'
+                          `catchIOError`
+                            \e ->
```

Comment:
It is only scoping over the `getMod`, only because I couldn't make it look pretty the other way :)

No point in catching the error and returning empty permissions to inspect - could instead just catch the exception outside of the permission check and return empty there.